tests/test_config.py

```diff
@@ -0,0 +1,299 @@
+# Copyright 2020-2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+import attr
+import pytest
+import sys
+
+from charmcraft.cmdbase import CommandError
+from charmcraft.config import (
+    _BasicPrime,
+    _CharmhubConfig,
+    _check_url,
+    load,
+)
+
+# Decide if we're using a Python 3.5 or older to support a jsonschema detail that uses
+# randomly ordered dictionaries
+is_py35 = (sys.version_info.major, sys.version_info.minor) < (3, 6)
```

Comment:
Just a nit, but I intuitively expected this to be `python == 3.5` or a compat level (py >= 3.5).

Reply:
I tried to code it generically like "3.5 or older", but then the name `is_py35` was more clear in the rest of the code. I can change it, I guess this will never run in py3.4 or 3.3, etc....